modules/core/src/main/java/org/locationtech/jts/geom/Coordinates.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (c) 2016 Vivid Solutions and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * and Eclipse Distribution License v. 1.0 which accompanies this distribution.
+ * The Eclipse Public License is available at http://www.eclipse.org/legal/epl-v10.html
+ * and the Eclipse Distribution License is available at
+ *
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ */
+package org.locationtech.jts.geom;
+
+/**
+ * Useful utility functions for handling Coordinate objects.
+ */
+public class Coordinates {
+  /**
+   * Factory method providing access to common Coordinate implementations.
+   * 
+   * @param dimension
+   * @return created coordinate
+   */
+  public static Coordinate create(int dimension)
+  {
+    return create(dimension, 0);
+  }
+
+  /**
+   * Factory method providing access to common Coordinate implementations.
+   * 
+   * @param dimension
+   * @param measures
+   * @return created coordinate
+   */
+  public static Coordinate create(int dimension, int measures)
+  {
+    if (dimension == 2) {
+      return new CoordinateXY();
+    } else if (dimension == 3 && measures == 0) {
+      return new Coordinate();
+    } else if (dimension == 3 && measures == 1) {
+      return new CoordinateXYM();
+    } else if (dimension == 4 && measures == 1) {
+      return new CoordinateXYZM();
+    }
+    return new Coordinate();
```

Comment:
Hmm...  This seems a tough case to handle properly...

Reply:
I think I should make a CoordinateSequence.createCoordiante() method, it will make the ExtendedCoordianteSequence example easier to work with as well. If not we are demanding that these subclasses be used which is poor design.